doc/man3/OSSL_HPKE_CTX_new.pod

```diff
@@ -0,0 +1,539 @@
+If required I<ikmlen> should be greater than or equal to
+OSSL_HPKE_get_recommended_ikmelen().
+
+OSSL_HPKE_decap() takes as input the sender's encapsulated public value
+produced by OSSL_HPKE_encap() (I<enc>) and the recipient's L<EVP_PKEY(3)>
+pointer (I<prov>), and then re-generates the internal secret derived by the
+sender. As before, an optional I<info> parameter allows binding that derived
+secret to other application/protocol artefacts. Only a single call to
+OSSL_HPKE_decap() is allowed for a given B<OSSL_HPKE_CTX>.
+
+OSSL_HPKE_open() is used by the recipient to decrypt the ciphertext I<ct> of
+size I<ctlen> using the I<ctx> and additional authenticated data I<aad> of
+size I<aadlen>, to produce the plaintext I<pt> of size I<ptlen>.
+On input I<ptlen> should contain the maximum size of the I<pt> buffer, and
+returns the output size. A I<pt> buffer that is the same size as the
+I<ct> buffer will suffice - generally the plaintext output will be
+a little smaller than the ciphertext input.
+An error will occur if the input I<ptlen> is too small.
+OSSL_HPKE_open() may be called multiple times, but as with OSSL_HPKE_seal()
+there is an internally incrementing nonce value so ciphertexts need to be
+presented in the same order as used by the OSSL_HPKE_seal().
+See L</Re-sequencing> if you need to process multiple ciphertexts in a
+different order.
+
+=head2 Exporting Secrets
+
+HPKE defines a way to produce exported secrets for use by the
+application.
+
+OSSL_HPKE_export() takes as input the B<OSSL_HPKE_CTX>, and an application
+supplied label I<label> of size I<labellen>, to produce a secret I<secret>
+of size I<secretlen>. The sender must first call OSSL_HPKE_encap(), and the
+receiver must call OSSL_HPKE_decap() in order to derive the same shared secret.
+
+Multiple calls to OSSL_HPKE_export() with the same inputs will produce the
+same secret.
+I<OSSL_HPKE_AEAD_ID_EXPORTONLY> may be used as the B<OSSL_HPKE_SUITE> I<aead_id>
+that is passed to OSSL_HPKE_CTX_new() if the user needs to produce a shared
+secret, but does not wish to perform HPKE encryption.
+
+=head2 Sender-authenticated HPKE Modes
+
+HPKE defines modes that support KEM-based sender-authentication
+B<OSSL_HPKE_MODE_AUTH> and B<OSSL_HPKE_MODE_PSKAUTH>. This works by binding
+the sender's authentication private/public values into the encapsulation and
+decapsulation operations. The key used for such modes must also use the same
+KEM as used for the overall exchange. OSSL_HPKE_keygen() can be used to
+generate the private value required.
+
+OSSL_HPKE_CTX_set1_authpriv() can be used by the sender to set the senders
+private I<priv> B<EVP_PKEY> key into the B<OSSL_HPKE_CTX> I<ctx> before calling
+OSSL_HPKE_encap().
+
+OSSL_HPKE_CTX_set1_authpub() can be used by the receiver to set the senders
+encoded pub key I<pub> of size I<publen> into the B<OSSL_HPKE_CTX> I<ctx> before
+calling OSSL_HPKE_decap().
+For EC curves, I<pub> is in an uncompressed encoding format.
```

Comment:
Hmm. Not sure,  I guess``OSSL_HPKE_CTX_set1_authpub()`` will work with a compressed form input but not sure if later calls will.  I forget if the uncompressed form will be recalculated internally so will have to make a test to check. I'll do that.

I guess, if we prefer to fail early, I could also make a check on the I<pub> provided too to see if it matches the suite and mode.

Note that use of compressed formats won't work for HPKE in general though because the uncompressed format has to be used internally for interop. There is an Internet-draft that specifies how to do that for HPKE but that's at an early stage (there's a [call for adoption](https://mailarchive.ietf.org/arch/msg/cfrg/dCxZkyF3zcFF9EJgXLJJmUfOags/) for the draft ongoing).

For the pod file, I've changed "is in an" to "should use an" which is likely better for the moment. I can clarify more when I've that test done. (Probably in a 2nd commit after the nits fixed.)

Reply:
So turns out you're correct that providing the compressed form public works there. Fair play for spotting that.

I deleted the line from the pod file that says otherwise.
I also added some more checking code to ``OSSL_HPKE_set1_authpub()`` that means we'll store the uncompressed form key in the context in any case, and added a test to check that the compressed form works.